hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxyImpl.java

```diff
@@ -1034,9 +1040,7 @@ public <R> Iterator<R> iterator(
             @Nonnull Projection<? super Map.Entry<K, V>, R> projection,
             @Nonnull Predicate<K, V> predicate
     ) {
-        if (predicate instanceof PagingPredicate) {
-            throw new IllegalArgumentException("Paging predicate is not allowed when iterating map by query");
-        }
+        PredicateUtils.checkDoesNotContainPagingPredicate(predicate, "iterator");
```

Comment:
minor: maybe we can update the javadoc to mention not supporting partition predicate that might contain a paging predicate